code/datums/riding.dm

```diff
@@ -249,6 +253,69 @@
 					buckled_mob.pixel_x = 10
 					buckled_mob.pixel_y = 5
 
+//prototype repair vehicle
+/datum/riding/space/repair
+	keytype = null
+	vehicle_move_delay = 1
+	var/obj/machinery/repair_turret/turret = null
+	var/obj/vehicle/space/speedbike/repair/R = null
+
+/datum/riding/space/repair/handle_vehicle_layer()
+	R = ridden
+	turret = R.turret
+	switch(ridden.dir)
+		if(NORTH)
+			ridden.pixel_x = -16
+			ridden.pixel_y = -16
+		if(SOUTH)
```

Comment:
I actually split it up for the turret before I took the turret layers out in an effort to get it working, ah well, ill just combine em like you asked.